src/auth/progress.auth.js

```diff
@@ -1,218 +1,139 @@
-            break;
-        case progress.data.Session.AUTH_TYPE_FORM_SSO:
-            authProv = new progress.data.AuthenticationProviderSSO(initObject.uri);
-            break;
-        case progress.data.Session.AUTH_TYPE_BEARER:
-            authProv = new progress.data.AuthenticationProviderBearer(initObject.uri);
-            break;
-        default:
-            // AuthenticationProvider: The 'init-object' parameter passed to the 'constructor' function
-            //                          has an invalid value for the 'authenticationModel' property.
-            throw new Error(progress.data._getMsgText(
-                "jsdoMSG502",
-                "AuthenticationProvider",
-                "init-object",
-                "constructor",
-                "authenticationModel"
-            ));
-            //break;
+    // load module base-64
+    try {
+        if (typeof btoa === "undefined") {
+            btoa = require("base-64").encode;
         }
+    } catch (exception3) {
+        console.error("Error: JSDO library requires btoa() function in Node.js.\n" +
+            "Please install base-64 package.");
+    }
+}());
+}
+
+authModel = initObject.authenticationModel.toLowerCase();
+switch (authModel) {
+    case progress.data.Session.AUTH_TYPE_ANON:
+        this._initialize(initObject.uri, progress.data.Session.AUTH_TYPE_ANON, { "_loginURI": progress.data.AuthenticationProvider._homeLoginURIBase });
+        authProv = this;
+        break;
+    case progress.data.Session.AUTH_TYPE_BASIC:
+        authProv = new progress.data.AuthenticationProviderBasic(initObject.uri);
+        break;
+    case progress.data.Session.AUTH_TYPE_FORM:
+        authProv = new progress.data.AuthenticationProviderForm(initObject.uri);
+        break;
+    case progress.data.Session.AUTH_TYPE_FORM_SSO:
+        authProv = new progress.data.AuthenticationProviderSSO(initObject.uri);
+        break;
+    default:
+        // AuthenticationProvider: The 'init-object' parameter passed to the 'constructor' function
+        //                          has an invalid value for the 'authenticationModel' property.
+        throw new Error(progress.data._getMsgText(
+            "jsdoMSG502",
+            "AuthenticationProvider",
+            "init-object",
+            "constructor",
+            "authenticationModel"
+        ));
+        //break;
+}
+
+return authProv;
+};
+
+
+// ADD METHODS TO THE AuthenticationProvider PROTOYPE
+
+// GENERIC IMPLEMENTATION FOR login METHOD THAT THE API IMPLEMENTATIONS OF login CAN CALL
+// (technically, they don't override it, they each have small login methods that call this)
+progress.data.AuthenticationProvider.prototype._loginProto =
+    function(sendParam) {
+        var deferred = new progress.util.Deferred(),
+            xhr,
+            uriForRequest,
+            header,
+            that = this;
 
-        return authProv;
-    };
-        
-
-    // ADD METHODS TO THE AuthenticationProvider PROTOYPE
```

Comment:
it looks like there are large code changes being made to this file. what is going on here?